main.py

```
@@ -18,7 +18,7 @@ def nth_fibonacci_util(n, memo):
 
     # Base case: if n is 0 or 1, return n
     if n <= 1:
-        return n
+        return m
```

Ruff (0.12.2)

Undefined name `m`

(F821)